tools/KoreBuild.Console/Commands/DockerBuildCommand.cs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace KoreBuild.Console.Commands
+{
+    internal class DockerBuildCommand : SubCommandBase
+    {
+        private const string DockerIgnore = ".dockerignore";
+        private const string DockerfileExtension = ".dockerfile";
+        private const string Owner = "aspnetbuild";
+        private const string ImageName = "korebuild";
+
+        public CommandArgument ImageVariant { get; set; }
+
+        public List<string> Arguments {get; set; }
+
+        public string Tag => $@"{Owner}/{ImageName}:{ImageVariant.Value}";
+
+        public override void Configure(CommandLineApplication application)
+        {
+            ImageVariant = application.Argument("image", "The docker image to run on.");
+            Arguments = application.RemainingArguments;
+
+            base.Configure(application);
+        }
+
+        protected override bool IsValid()
+        {
+            if(string.IsNullOrEmpty(ImageVariant?.Value))
+            {
+                Reporter.Error("Image is a required argument.");
+                return false;
+            }
+
+            return true;
+        }
+
+        protected override int Execute()
+        {
+            var dockerFileName = GetDockerFileName(ImageVariant.Value);
+            var dockerFileSource = GetDockerFileSource(dockerFileName);
+            var dockerFileDestination = Path.Combine(RepoPath, GetDockerFileName(ImageVariant.Value));
+
+            File.Copy(dockerFileSource, dockerFileDestination, overwrite: true);
+
+            var dockerIgnoreSource = GetDockerFileSource(DockerIgnore);
+            var dockerIgnoreDestination = Path.Combine(RepoPath, DockerIgnore);
+
+            File.Copy(dockerIgnoreSource, dockerIgnoreDestination, overwrite: true);
+
+            // If our ToolSource isn't http copy it to the docker context
+            var dockerToolsSource = ToolsSource;
+            string toolsSourceDestination = null;
+            if (!ToolsSource.StartsWith("http"))
+            {
+                dockerToolsSource = "ToolsSource";
+                toolsSourceDestination = Path.Combine(RepoPath, dockerToolsSource);
+                DirectoryCopy(ToolsSource, toolsSourceDestination);
```

Comment:
An easier and faster way to do this to volume mount this folder into the container.